src/data/finset/prod.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2015 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura, Jeremy Avigad, Minchao Wu, Mario Carneiro
```

Comment:
It might be worth re-evaluating this author list from a git blame

Reply:
For instance, `finset.product` is attributed to Johannes in the header of the file it originated from:

https://github.com/leanprover-community/mathlib/blob/c03c16d6fb81d23761227a56e6f39d7baddb9bc1/data/finset/fold.lean#L95-L99

Reply:
And `off_diag` and `diag` seemed to come from #4502

Reply:
```suggestion
Authors: Johannes Hölzl, Mario Carneiro, Oliver Nash
```

Reply:
(Minchao's contribution was 79ed1c390dff0b4c5b77f726d5d414c948ab76cf which isn't one of the moved lemmas)

Reply:
Wow, thanks for tracking this! I usually review the authors list but since I've worked on `data.list.basic` I started assuming basically all the file had been a monolith coming straight out of Microsoft.

Also note that Minchao did more than that: mostly #15 but you can see [the full list](https://github.com/leanprover-community/mathlib/commits?author=minchaowu&before=211bdff0f018fe870efe817e13900cd2e481653b+35), but still not about `finset.product`.